### ISSUE TITLE
error while issuing request: Post "http://localhost:23000/gitpod/cli"

### ISSUE THREAD

**Issue by nisarhassan12** [gitpod-io/gitpod#2041]
### Describe the bug

The port did not seem to be working I was getting port 8000 didn't respond which does not usually happen in a Gatsby project. As a workaround I have added the following:

```yml
- command: gp await-port 8000 && sleep 3 && gp preview $(gp url 8000)
```

which seems to be failing with the following error:

![image](https://user-images.githubusercontent.com/46004116/95669206-4a553f00-0b97-11eb-8dee-6dccc8894588.png)

Ref: https://github.com/TypeFox/gitpodify/issues/476#issuecomment-706641895

### Steps to reproduce

Open a workspace for https://github.com/nisarhassan12/svgr

The issue can be re-produced on both staging and production.

**Comment by akosyakov** [comment]
@nisarhassan12 Can be closed?

**Comment by nisarhassan12** [comment]
@akosyakov Yeah. I'll do it.

**Comment by ashmckenzie** [comment]
I think this issue may have re-surfaced. When I follow the steps to reproduce, I see the same error on a fresh instance just now:

```sh
gitpod /workspace/svgr $  HISTFILE=/workspace/.gitpod/cmd-0 history -r; {
> gp await-port 8000 && sleep 3 && gp preview $(gp url 8000)
> }
Awaiting port 8000... ok
2021/02/16 07:30:06 error while issuing request: Post "http://localhost:23000/gitpod/cli": context deadline exceeded (Client.Timeout exceeded while awaiting headers)
```

If I wait a few moments and re-run `gp await-port 8000 && sleep 3 && gp preview $(gp url 8000)`, I see no error though:

```sh
gitpod /workspace/svgr $ gp await-port 8000 && sleep 3 && gp preview $(gp url 8000)
Awaiting port 8000... ok
```

**Comment by corneliusludmann** [comment]
Hi @ashmckenzie!

The error message

> error while issuing request: Post "http://localhost:23000/gitpod/cli": context deadline exceeded (Client.Timeout exceeded while awaiting headers)

is usually misleading (we need to fix this). Despite this error message the command is executed as expected. Do you observe any symptoms/problems that lead you to the assumption that something actually goes wrong?

**Comment by ashmckenzie** [comment]
@corneliusludmann thanks for the quick response 🙂

> Do you observe any symptoms/problems that lead you to the assumption that something actually goes wrong?

Everything seems to work OK, other than a few people mentioning the error and causing confusion 🙂 Would be great to have this fixed ASAP, please ping me on the Issue that's tracking the work, thanks! :+1:

**Comment by apolopena** [comment]
I have noticed  that my terminals seems to get blocked by an apparent synchronous action but I haven't looked deep into the root cause yet. This bug seems to happen only when the workspace is stopped and then started again. My calls to `gp preview` are in a `command` task.

**Comment by jjjack1** [comment]
anyone help me to connect to my own url? not gp url

**Comment by pawlean** [comment]
> anyone help me to connect to my own url? not gp url

As in local host? https://www.gitpod.io/docs/editors/vscode

**Comment by jjjack1** [comment]
> > anyone help me to connect to my own url? not gp url
> 
> As in local host? https://www.gitpod.io/docs/editors/vscode

I bought url domain for anyone who can be accessed to my homepage

**Comment by akosyakov** [comment]
@jjjack1 Could you create a new issue please? And provide more details. What are you doing, what do you expect and what goes wrong pelase.
